core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -3167,6 +3170,687 @@ public void testRegisterExistingTable() {
     assertThat(catalog.dropTable(identifier)).isTrue();
   }
 
+  @Test
+  public void testCreateBranchFromCurrentSnapshot() {
+    C catalog = catalog();
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    Table table = catalog.buildTable(TABLE, SCHEMA).create();
+    table.newFastAppend().appendFile(FILE_A).commit();
+    Snapshot currentSnapshot = table.currentSnapshot();
+
+    String branchName = "data-quality-check";
+    table.manageSnapshots().createBranch(branchName).commit();
+
+    // Verify branch was created
+    Map<String, SnapshotRef> refs = table.refs();
+    assertThat(refs).containsKey(branchName);
+    assertThat(refs.get(branchName).isBranch()).isTrue();
+    assertThat(refs.get(branchName).snapshotId()).isEqualTo(currentSnapshot.snapshotId());
+  }
+
+  @Test
+  public void testCreateBranchFromSpecificSnapshot() {
```

Comment:
I would probably just have one of these tests. In the context of CatalogTests, we're really just trying to test if a catalog allows for new branches.